nix/README.md

```diff
@@ -0,0 +1,63 @@
+# Nix-based kubernetes test environment
+
+All operations are based on having the `nix` package manager available. It can be easily be installed using the quickstart instructions found at https://nixos.org/download.html#nix-quick-install, which are just
+
+```console
+$ curl -L https://nixos.org/nix/install | sh
+```
+
+## Building a docker image of a node
+
+To build your current source into a container image, enter a nix-shell in the same folder as this `README.md`, then run `build-node.sh`.
+
+```console
+$ nix-shell
+$ ./build-node.sh
+[...]
+Created new docker image casper-node:f2b9cd7a-dirty.
+
+Load into local docker:
+docker load -i ./result
+
+Publish image
+skopeo --insecure-policy copy docker-archive:./result docker://clmarc/casper-node:f2b9cd7a-dirty
```

Comment:
A few things:

  - Should we really be using `clmarc` on docker-hub?
  - Why `insecure-policy`?  Maybe there's no way around it...
  - Rather than have `skopeo` in `shell.nix`,  I think it's better to say:

```console
nix-shell --pure -p skopeo --command "skopeo --insecure-policy copy docker-archive:./result docker://clmarc/casper-node:f2b9cd7a-dirty"
```

Reply:
> Should we really be using clmarc on docker-hub?

Definitely not. This is due change soon.

> Why insecure-policy? Maybe there's no way around it...

It has zero effect, as far as I know. We're uploading a local file to a remote repo, no docker policies are involved.

> Rather than have skopeo in shell.nix, I think it's better to say: [...]

Possibly, but a lot less newbie friendly :) Arguably it's not even good to have these commands **echo**'d instead of offering a more integrated tool that does with a simple `upload` or even an `--upload` flag to `build-image.sh`.

This is all due to the fact that this tooling is currently in a lot of flux still.